atef/procedure.py

```diff
@@ -81,27 +159,142 @@ class DeviceConfiguration:
 class ConfigurationCheckStep(ProcedureStep):
     """Step which checks device configuration versus a given timestamp."""
     #: Device name to device configuration information.
-    devices: Dict[str, DeviceConfiguration]
+    devices: Dict[str, DeviceConfiguration] = field(default_factory=dict)
 
 
 @dataclass
 class TyphosDisplayStep(ProcedureStep):
     """A procedure step which opens one or more typhos displays."""
     #: Happi device name to display options.
-    devices: Dict[str, DisplayOptions]
+    devices: Dict[str, DisplayOptions] = field(default_factory=dict)
 
 
 @dataclass
 class PydmDisplayStep(ProcedureStep):
     """A procedure step which a opens a PyDM display."""
     #: The display path.
-    display: pathlib.Path
+    display: pathlib.Path = field(default_factory=pathlib.Path)
     #: Options for displaying.
-    options: DisplayOptions
+    options: DisplayOptions = field(default_factory=DisplayOptions)
+
+
+@dataclass
+class PassiveStep(ProcedureStep):
+    """A step that runs a passive checkout file"""
+    filepath: pathlib.Path = field(default_factory=pathlib.Path)
+
+    def _run(self) -> Result:
+        """ Load, prepare, and run the passive step """
+        if self.filepath.suffix == '.json':
```

Comment:
Not required in this PR, but a refactor thought/suggestion: `ConfigurationFile` should do the filename check and use the appropriate loader

-> `ConfigurationFile.from_filename(self.filepath)`